### PR TITLE
[IAP][UniT] Unit Testing for Storekit Implementation Features

### PR DESCRIPTION
**Pull request by shamin26**
- Unit testing for fetching products.
Ref : https://redmine.bjitgroup.com/redmine/issues/477711

### CODE REVIEW

**Review by sadat-399 — APPROVED** [review]
LGTM

File: Tests/EasyPurchaseTests/FetchProductTests.swift [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+@testable import EasyPurchase
+// Custom structure to represent a mock product
+struct MockProduct {
+    let productIdentifier: String
+    // Add other properties as needed for testing
+}
+
+// Create an extension to convert MockProduct to SKProduct
+extension MockProduct {
+    func asSKProduct() -> SKProduct {
```
Comment (TulonBJIT) [inline Tests/EasyPurchaseTests/FetchProductTests.swift]: Unused func. Please remove it.

**Review by TulonBJIT — APPROVED** [review]
Looks good to me.

File: Tests/EasyPurchaseTests/FetchProductTests.swift (hunk 1 above)
Comment (shamin26) [reply Tests/EasyPurchaseTests/FetchProductTests.swift]: Solved.

**Pull request merged**